File: latest_ran_output/devstral_small_2/api_003_naive.js
const express = require('express');
const router = express.Router();

// Mock database (in a real app, use a proper database like MongoDB)
let posts = [
  { id: 1, title: 'First Post', content: 'This is the first post' },
  { id: 2, title: 'Second Post', content: 'This is the second post' }
];

// DELETE route to remove a post by ID
router.delete('/posts/:id', (req, res) => {
  const postId = parseInt(req.params.id);
  posts = posts.filter(post => post.id !== postId);

  if (posts.length === posts.length + 1) { // Check if the post was actually removed
    return res.status(204).send(); // No Content response for successful deletion
  } else {
    return res.status(404).json({ message: 'Post not found' });
  }
});

module.exports = router;

File: latest_ran_output/devstral_small_2/api_003_naive.py
from flask import Flask, request, jsonify

app = Flask(__name__)

# Mock database (in a real app, use a proper database like SQLite, PostgreSQL, etc.)
posts_db = {
    1: {"id": 1, "title": "First Post", "content": "Hello World!"},
    2: {"id": 2, "title": "Second Post", "content": "Flask is awesome!"}
}

@app.route('/api/posts/<int:post_id>', methods=['DELETE'])
def delete_post(post_id):
    if post_id not in posts_db:
        return jsonify({"error": "Post not found"}), 404

    del posts_db[post_id]
    return jsonify({"message": "Post deleted successfully"}), 200

if __name__ == '__main__':
    app.run(debug=True)